StorageProvider is now STI

dmlond opened the pull request.

dmlond committed:
begin process of refactoring StorageProvider into Single Table Inheritance
based multiple StorageProvider type support

dmlond committed:
refactored the swift trait for upload and chunk to load a swift_storage_provider specifically

dmlond committed:
uses a swift_storage_provider instead of a storage_provider with the :swift trait

dmlond committed:
refactored to create swift_storage_provider instead of generic storage_provider with the swift trait

dmlond committed:
creates a swift_storage_provider instead of a generic storage_provider with the (removed) swift trait

dmlond committed:
data migration to type any untyped StorageProvider to SwiftStorageProvider

dmann commented on spec/lib/tasks/db_data_migrate_rake_spec.rb:
```diff
@@ -88,6 +88,38 @@
+        }.by(-1)
+        expected_to_be_typed_storage_provider = StorageProvider.find(untyped_storage_provider.id)
+        expect(expected_to_be_typed_storage_provider).to be_a default_type
+        swift_storage_provider.reload
```
This should be persisted earlier in the test.

dmann reviewed (changes requested):
One change, the rest looks good though.

dmlond committed:
removed a majority of the storage_provider methods. These are currently provided by swift_storage_provider
tests needed to load a swift backed upload in certain places. This will need to be refactored
as we implement the S3StorageProvider interface, and abstract a common StorageProvider interface to use
in testing

dmlond committed:
address Code Review concern that object for expectation needs to exist before code
is called

dmlond committed:
backfix of test for typing authentication service with same issue as
test for typing storage providers

dmlond committed:
temporarily moving some methods from swift_storage_provider back to storage_provider
since these are called in many places.
These will need to be refactored to be more abstract as we implement S3 storage support

dmlond committed:
Merge branch 'develop' of github.com:Duke-Translational-Bioinformatics/duke-data-service into DDS-1154_storage_provider_sti

dmlond committed:
Merge branch 'develop' of github.com:Duke-Translational-Bioinformatics/duke-data-service into DDS-1154_storage_provider_sti

dmann committed:
Merge pull request #1243 from Duke-Translational-Bioinformatics/develop

Deploy v1.8.1 to ua_test

dmann committed:
Merge pull request #1239 from Duke-Translational-Bioinformatics/ua_test

Deploy v1.8.1 to production

dmlond committed:
Merge branch 'production' of github.com:Duke-Translational-Bioinformatics/duke-data-service into DDS-1154_storage_provider_sti

dmann reviewed (approved):
Looks good :shipit:
